include/boost/geometry/io/wkt/write.hpp

```diff
@@ -62,22 +63,67 @@ namespace boost { namespace geometry
 namespace detail { namespace wkt
 {
 
+template <typename Output>
+struct output_formatter
+{
+    template <typename T>
+    static void append(Output& out, T const& v)
+    {
+        out << v;
+    }
+};
+
+template <typename CharT, typename Traits, typename Allocator>
+struct output_formatter<std::basic_string<CharT, Traits, Allocator>>
+{
+    template <typename T>
+    static void append(std::basic_string<CharT, Traits, Allocator>& out, T const& val)
+    {
+        std::basic_stringstream<CharT, Traits, Allocator> ss;
+        ss << val;
```

Comment:
As I mentioned here: https://github.com/boostorg/geometry/issues/654#issuecomment-570940504 I propose to set the precision here to `std::numeric_limits<T>::digits10` if `std::numeric_limits<T>::is_specialized`. @mloskot what do you think?

Reply:
I agree maxing out by `T` should be fine.

We could have something like this to address user requests like https://github.com/boostorg/geometry/issues/654#issuecomment-570835846

```
std::string to_wkt(Geometry const& geometry); // defaults to digits10
std::string to_wkt(Geometry const& geometry, int significant_digits);

Reply:
Can you provide any technique to propagate the significant_digits back to the output_formatter?
The ways that I see its possible are 
- Use a tparam or param in each function call from to_wkt till output_formatter
- Use a global variable
Both of these seem to be bad. The latter worse than the former. If you could provide any other way or else I will proceed with the first one.

Reply:
@sudo-panda 
> Use a tparam or param in each function call from to_wkt till output_formatter

IMO, users typically expect it to be specifiable  at run-time, so yes, adding the parameter is the feasible option `output_formatter<T>::append(..., int significant_digits)` .

> Use a global variable

No, we definitely should avoid globals.

> Both of these seem to be bad.

What's bad in an extra parameter?

/cc @vissarion, @awulkiew for insights

Reply:
> What's bad in an extra parameter?

It's redundant information for the functions that are between to_wkt and output_formatter. But if that's the only way then I will do it.

Reply:
The existing technique of dispatching, often interleaved (for geometry type, for output type), is based on forwarding of (numerous) parameters which are often ignored at higher levels until they reach place where they find its use. Think of it as functional way.

Alternative would be object-based solution, for example:

```cpp
template <typename Geometry>
inline std::string to_wkt(Geometry const& geometry, int signigicant_digits)
{
    concepts::check<Geometry const>();
    std::string out;
    wkt<dispatch::devarianted_wkt<Geometry>> wkt(out, signigicant_digits);
    wkt.write(out, geometry);
    return out;
}
```

but then `wkt` and others would have to undergo a major re-write.

To me, this is a matter of work-to-feature balance. Adding support for how many decimal digits we output may not necessarily justify a major change to existing code. Major changes have real potential to introduce bugs. Is the feature worth the risk...

Reply:
You are right it is not worth the risk and is not how the rest of boost/geometry has been built. I will start work on adding the parameter.

Reply:
If you are not in rush, you may give @awulkiew  and @vissarion some time to respond with their opinions.

Reply:
Sure

Reply:
To my point of view, passing an extra parameter seems ok and as @mloskot said it is already performed in the library in many places.

Reply:
I'm also ok with an extra parameter.

The only problem I see is that a user may want to pass a set of such parameters, i.e. stream manipulators for coordinates. But this problem is solvable by e.g. allowing to optionally pass a `std::pair` or `std::tuple` of manipulators in the future:

    bg::to_wkt(geometry);
    bg::to_wkt(geometry, int_precision_10);
    bg::to_wkt(geometry, std::make_pair(std::fixed, std::setprecision(10)));

unless you have a better idea of an extensible interface. Just have in mind that in the future we may have to allow passing a strategy into this function as with any other function.

Reply:
Why not just let user give own pre-configured stream?
```
to_wkt(std::ostream& os, geometry);
```